app/services/hyrax/solr_query_service.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+module Hyrax
+  ##
+  # Methods in this class are providing functionality previously supported by ActiveFedora::SolrQueryBuilder.
+  # It includes methods to build and execute a query.
+  class SolrQueryService
+    attr_reader :query, :solr_service
+
+    def initialize(query: [], solr_service: Hyrax::SolrService)
+      @query = query
+      @solr_service = solr_service
+    end
+
+    ##
+    # @return [Hash] the results returned from solr for the current query
+    def get
+      solr_service.get(build)
+    end
+
+    ##
+    # @return [Integer] the number of results that match the query in solr
+    def count
+      solr_service.count(build)
+    end
+
+    ##
+    # @return [String] the combined query that can be submitted to solr
+    def build
+      return 'id:NEVER_USE_THIS_ID' if @query.blank? # forces this method to always return a valid solr query
+      @query.join(' AND ')
+    end
+
+    ##
+    # @return [Hyrax::SolrQueryService] the existing service with the query reset to empty
+    def reset
+      @query = []
+      self
+    end
+
+    ##
+    # @param ids [Array] id_array the ids that you want included in the query
+    # @return [Hyrax::SolrQueryService] the existing service with id query appended
+    def with_ids(ids: [])
+      raise ArgumentError, "Expected [Array] for ids:, but received <#{ids.class}>" unless ids.is_a? Array
+      ids = ids.reject(&:blank?)
+      raise ArgumentError, "Expected there to be at least one non-blank id." if ids.blank?
+      id_query = construct_query_for_ids(ids)
+      @query += [id_query]
+      self
+    end
+
+    ##
+    # @param model [Class|String] a class from the model (e.g. Hyrax::Work, Hyrax::FileSet, etc.)
+    # @return [SolrQueryService] the existing service with model query appended
+    def with_model(model:)
+      raise ArgumentError, "Expected <Class> or <String> for model:, but received <#{model.class}>" unless model.is_a?(Class) || model.respond_to?(String)
```

Comment:
i think `model.respond_to?(String)` might be an error?

it seems like the argument type for `model` is `#to_s`? could the whole type check be dropped?